.github/workflows/build.yaml

```diff
@@ -206,12 +206,26 @@ jobs:
 
     runs-on: ubuntu-latest
 
+    if: startsWith(github.ref, 'refs/tags/')
+
     steps:
     - uses: actions/checkout@v4
 
     - name: Download all artifacts
       uses: actions/download-artifact@v4
 
+    - name: Rename artifacts
+      run: |
+        mv vcpkg-x64-*-${{ github.sha }}/vcpkg-x64-windows-2019-${{ github.sha }}.7z windows-2019-${{ github.ref_name }}.7z
+        mv vcpkg-x64-*-${{ github.sha }}/vcpkg-x64-windows-2022-${{ github.sha }}.7z windows-2022-${{ github.ref_name }}.7z
+        mv vcpkg-x64-*-${{ github.sha }}/vcpkg-x64-windows-2019-pdb-${{ github.sha }}.7z windows-2019-pdb-${{ github.ref_name }}.7z
+        mv vcpkg-x64-*-${{ github.sha }}/vcpkg-x64-windows-2022-pdb-${{ github.sha }}.7z windows-2022-pdb-${{ github.ref_name }}.7z
+
+    - name: Release
+      uses: softprops/action-gh-release@v2
+      with:
+        files: windows-*-${{ github.ref_name }}.7z
```

Comment:
With what github will populate this? I assume this should happen only on PR merge to master. What will be a value of `github.ref_name` in this case?

Reply:
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#github-context

Since the job only runs for tags, `ref_name` will always be the name of the tag.

Reply:
This means the job is not checked by default in PRs. It's designed to do everything except actual push to a remote repo for PRs. This can stay the same except not making a release if it's not a tag job.